flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/expressions/ExpressionParser.scala

```diff
@@ -59,6 +59,8 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
           Literal(str.toInt)
         } else if (str.endsWith("f") | str.endsWith("F")) {
           Literal(str.toFloat)
+        } else if (str.endsWith(".")) {
```

Comment:
Of course both are valid expressions but `1.` is simply not a valid number literal. Instead `1` would be valid. I guess you should change your grammar such that `1.abs()` will parsed into something like `functionApplication(NumberLiteral(1), abs)` and the `.` indicates the function application but is not part of the number literal token.